src/gui/guiFormSpecMenu.cpp

```diff
@@ -445,105 +459,95 @@ void GUIFormSpecMenu::parseScrollContainerEnd(parserData *data)
 
 void GUIFormSpecMenu::parseList(parserData *data, const std::string &element)
 {
-	if (m_client == 0) {
-		warningstream<<"invalid use of 'list' with m_client==0"<<std::endl;
-		return;
-	}
+	MY_CHECKCLIENT("list");
```

Comment:
Can't this check go elsewhere?
The client reference being undefined seems like such an obvious error that it should even get this far.

Reply:
So I thought too until I noticed that the Main Menu uses GUIFormSpec for rendering.

Reply:
That's unfortunate.